Default to system font in attribution titles

1ec5 opened the pull request:
The inline stylesheet used when converting an attribution HTML string to an attributed string now specifies the system font as the default font. It still isn’t clear to me why macosapp has been defaulting to the system font, but it makes sense to me that the default default font would be Times without this change. This change makes use of WebKit’s proprietary `-apple-system` keyword, which [Apple recommends](https://webkit.org/blog/3709/using-the-system-font-in-web-content/) for this purpose.

There is still a slight difference between macosapp and an application outside macos.xcworkspace, but I think the two results are close enough:

<img width="239" alt="fonts" src="https://cloud.githubusercontent.com/assets/1231218/21957417/76aa3d30-da4a-11e6-815b-f0acfbee77d3.png">

Fixes #7722.

/cc @frederoni @friedbunny

1ec5 commented:
This issue is tagged as a release blocker because it’s a highly visible regression on macOS. It is not a release blocker on iOS, as it affects iOS only to the extent that a developer would try to display an MGLAttributionInfo title as rich text. The attribution sheet is unaffected because it displays each title as plain text. Even so, I’d like to land this change in time for macOS SDK v0.3.0, which means it’ll be in iOS SDK v3.4.0 as well.

friedbunny commented:
The [WebKit blog says](https://webkit.org/blog/3709/using-the-system-font-in-web-content/) `-apple-system` is supported from OS X 10.11 (and iOS 9, incidentally). We support back to 10.10, so I think it’d make sense to also add `"Helvetica Neue"` to this rule, the previous system font.

The only question I would then have is if OS X 10.10 still used Helvetica Neue at small sizes...

1ec5 commented:
> The only question I would then have is if OS X 10.10 still used Helvetica Neue at small sizes...

Yes, Lucida Grande was supplanted by Helvetica Neue at all three control sizes.

1ec5 commented:
Support for the `-apple-system` keyword was introduced in WebKit/webkit@a03afe285974b0ee0d3dcd0fc017d7ac21359e7c for [bug 144707](https://bugs.webkit.org/show_bug.cgi?id=144707), which made it into iOS 9 and OS X 10.10. But the same functionality was actually available earlier via the identical `-apple-system-font` keyword, upstreamed from iOS in WebKit/webkit@b8ade587ad740cb2837293d8bc77cd22fe82bb7b for [bug 126654](https://bugs.webkit.org/show_bug.cgi?id=126654) (which would’ve made it into [iOS 8](https://trac.webkit.org/browser/tags/Safari-538.35/Source/WebCore/platform/graphics/ios/FontCacheIOS.mm#L570)) and introduced to OS X in WebKit/webkit@989ba141ffea9573061d337e73bea17467399a99 for [bug 133382](https://bugs.webkit.org/show_bug.cgi?id=133382) (which made it into [Safari 538.1](https://trac.webkit.org/browser/tags/Safari-538.1/Source/WebCore/platform/mac/WebFontCache.mm) and thus [OS X 10.10](https://en.wikipedia.org/wiki/Safari_version_history#Mac)). Therefore I’ve added `-apple-system-font` as a fallback. I also added `sans-serif` to at least ensure something reasonably close to Helvetica Neue on iOS 7.

Additionally, on macOS, I set `font: -webkit-mini-control` to apply a font identical to what macosapp displays:

<img width="231" alt="font" src="https://cloud.githubusercontent.com/assets/1231218/21958634/928ce21a-da67-11e6-81e0-cca55087ebe7.png">

friedbunny commented:
@1ec5 You’re absolutely correct — the blog piece I cited doesn’t actually say when `-apple-system` became available, just that you could use San Francisco via that keyword in OS X 10.11 and iOS 9. :bow:
